htsget-search/src/storage/local.rs

```diff
@@ -72,7 +72,10 @@ impl Storage for LocalStorage {
 
     // TODO file:// is not allowed by the spec. We should consider including an static http server for the base_path
     let path = self.get_path_from_key(key)?;
-    let url = Url::new(format!("file://{}", path.to_string_lossy())).with_headers(headers);
+    let url = match options.class {
+      None => Url::new(format!("file://{}", path.to_string_lossy())).with_headers(headers),
+      Some(class) => Url::new(format!("file://{}", path.to_string_lossy())).with_headers(headers).with_class(class)
+    };
```

Comment:
Could you please move the common part into an intermediate variable?

```suggestion
    let url = Url::new(format!("file://{}", path.to_string_lossy())).with_headers(headers);
    let url = match options.class {
      Some(class) => url.with_class(class),
      None => url,
    };
    Ok(url)
```